CorsixTH/Lua/epidemic.lua

```diff
@@ -261,16 +262,22 @@ function Epidemic:checkNoInfectedPlayerHasLeft()
     if infected_patient.going_home and not infected_patient.cured and
         px and py and not self.hospital:isInHospital(px,py) then
       -- Patient escaped from the hospital, discovery is inevitable.
-      self.result_determined = true
-      if not self.inspector then
-        self:spawnInspector()
-      end
       self:finishCoverUp()
       return
     end
   end
 end
 
+function Epidemic:checkThereAreNoAnyInfectedPatientsLeft()
+  if self.result_determined then return end
+
+  local no_any_infected_patients_left = self:countInfectedPatients() == 0
+  if no_any_infected_patients_left then
```

Comment:
You could move the `self:countInfectedPatients() == 0` check inside the `if`

Having zero infected patient implies already they don't exist (which is what I read in the variable name).

Reply:
change to `if (self:countInfectedPatients() == 0) then`